crypto/x509/x509_lu.c

```diff
@@ -647,9 +649,14 @@ STACK_OF(X509) *X509_STORE_get1_all_certs(X509_STORE *store)
     return NULL;
 }
 
-/* Returns NULL on internal/fatal error, empty stack if not found */
-STACK_OF(X509) *X509_STORE_CTX_get1_certs(X509_STORE_CTX *ctx,
-                                          const X509_NAME *nm)
+/*-
+ * Collect from |ctx->store| all certs with subject matching |nm|.
+ * Caller must free on ret == 1 the resulting list |*certs|.
+ * Caller must also free its entries if up_refs != 0.
+ * Returns NULL on internal/fatal error, empty stack if not found.
+ */
+STACK_OF(X509) *ossl_x509_store_ctx_get_certs(X509_STORE_CTX *ctx,
+                                              const X509_NAME *nm, int up_refs)
```

Comment:
Does the `up_refs == 0` case really  make sense as an API?  Without the ref count bump the returned objects may be invalid by the time the caller looks at them. What is the use-case for `up_refs == 0`?

Reply:
Oh, right, this is not thread-safe.
It was meant as minor optimisation, but not a good idea.
Reverted this to `X509_STORE_CTX_get1_certs()` only.